src/dawn/IIR/IIR.h

```diff
@@ -25,7 +25,7 @@ namespace iir {
 /// @ingroup optimizer
 class IIR : public IIRNode<void, IIR, Stencil> {
 
-  const std::array<unsigned int, 3> blockSize_ = {{32, 4, 4}};
+  std::array<unsigned int, 3> blockSize_ = {{32, 4, 4}};
```

Comment:
this should be derived info, right? (since we're deriving it in `passSetBlockSize`)

Reply:
yes and no, the pass is computing a default (in case the user does not specify anything), but in the future we should be able to not want this and set it in the IIR ourselves (similar argument to caches). The information is computed using a strategy but not really derived deterministicylly from the information in the IIR (is more like a choice)

Reply:
I see your point to some extent. Thought I would argue that all the information that is generated in the optimizer-process in dawn should be derived and everything that we get from the HIR should be non-derived. If you derive the info from a command-line argument or via smart algorithmic decisions is not really relevant is it?

Reply:
not really, plenty of information encoded in the IIR is a result of optimizers and is not derived. Examples are the way the computations are structured in stages, ms, etc. caches, computations on the fly or not, etc. 

What is derived is information that is redundant and can be derived from IIR (itself) without algorithmic choices ?